tsl/src/compression/compression.c

```diff
@@ -2355,7 +2355,7 @@ decompress_batches_for_update_delete(List *chunks, List *predicates)
 	SegmentFilter *filter;
 
 	if (predicates)
-		fill_predicate_context(linitial(chunks), predicates, &filters, &is_null);
```

Comment:
I agree with @kgyrtkirk please do not use nested lists here. Use proper struct if needed or act on the information straight away since only use this information once after and dont need to keep it.

Reply:
Now i iterate over chunks and its corresponding predicates and build scan keys and process the chunks.